modules/objdetect/src/qrcode.cpp

```diff
@@ -988,10 +992,26 @@ class QRDecode
     std::string getDecodeInformation() { return result_info; }
     bool straightDecodingProcess();
     bool curvedDecodingProcess();
+    vector<Point2f> alignment_coords;
+    vector<Point2f> getOriginalPoints() {return original_points;}
 protected:
     double getNumModules();
-    bool updatePerspective();
+    Mat getHomography() {
+        CV_TRACE_FUNCTION();
+        const Point2f centerPt = intersectionLines(original_points[0], original_points[2],
+                                               original_points[1], original_points[3]);
+        if (cvIsNaN(centerPt.x) || cvIsNaN(centerPt.y))
+            return Mat();
```

Comment:
Looks like the check is redundant. In case if the points have wrong topology, the next findHomography returns empty matrix as expected.